python/ray/tests/test_autoscaler_yaml.py

```diff
@@ -601,6 +601,22 @@ def mock_list_subnets(*args):
             "ray_head_default"]["node_config"][
                 "networkInterfaces"] == default_interfaces
 
+    @pytest.mark.skipif(
+        sys.platform.startswith("win"), reason="Fails on Windows.")
+    def testResourceValidationFailure(self):
+        """Validates that resource field is correctly validated."""
+        path = os.path.join(RAY_PATH, "autoscaler", "aws", "example-full.yaml")
+        config = yaml.safe_load(open(path).read())
+        node_type = config["available_node_types"]["ray.head.default"]
+        # Invalid `resources` field, say user entered `resources: `.
+        node_type["resources"] = None
+        with pytest.raises(jsonschema.exceptions.ValidationError):
+            validate_config(config)
+        # Invalid value in resource dict.
+        node_type["resources"] = {"CPU": "a string is not valid here"}
+        with pytest.raises(jsonschema.exceptions.ValidationError):
+            validate_config(config)
```

Comment:
Is this testing the change to the JSON schema?

Reply:
Yep, demonstrating the bug fix.
This test (first `with block` specifically) would fail in current master.